scripts/mod_loader/modapi/board.lua

```diff
@@ -1,96 +1,691 @@
+		self:SetAcid(loc, false)
+		self:SetTerrain(loc, TERRAIN_ROAD)
+	end
+end
 
-	local dmg = SpaceDamage(loc)
-	dmg.iFire = fire and EFFECT_CREATE or EFFECT_REMOVE
-	self:DamageSpace(dmg)
+BoardClass.GetHealth = function(self, loc)
+	Tests.AssertSignature{
+		ret = "int",
+		func = "GetHealth",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.GetTileHealth(self, loc)
+end
 
-	self:RestorePawnsToTile(loc, pawnStack)
+BoardClass.GetMaxHealth = function(self, loc)
+	Tests.AssertSignature{
+		ret = "int",
+		func = "GetMaxHealth",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.GetTileMaxHealth(self, loc)
 end
 
-BoardClass.SetShield = function(self, loc, shield)
-	Tests.AssertEquals("userdata", type(self), "Argument #0")
-	Tests.AssertTypePoint(loc, "Argument #1")
-	Tests.AssertEquals("boolean", type(shield), "Argument #2")
+BoardClass.GetLostHealth = function(self, loc)
+	Tests.AssertSignature{
+		ret = "int",
+		func = "GetLostHealth",
+		params = { self, loc },
+		{ "userdata|GameBoard&", "userdata|Point" }
+	}
+	
+	return CUtils.GetTileLostHealth(self, loc)
+end
 
-	local dmg = SpaceDamage(loc)
-	dmg.iShield = shield and EFFECT_CREATE or EFFECT_REMOVE
-	self:DamageSpace(dmg)
+BoardClass.SetHealth = function(self, loc, hp)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetHealth",
+		params = { self, loc, hp },
+		{ "userdata|GameBoard&", "userdata|Point", "number|int" }
+	}
+	
+	local hp_max = CUtils.GetTileMaxHealth(self, loc)
+	local iTerrain = self:GetTerrain(loc)
+	hp = math.max(0, math.min(hp, hp_max))
+	
+	local rubbleState = CUtils.GetTileRubbleState(self, loc)
+	local isRubble = iTerrain == TERRAIN_RUBBLE
+	local isBuilding = iTerrain == TERRAIN_BUILDING
+	local isMountain = iTerrain == TERRAIN_MOUNTAIN
+	local isRuins = isRubble and rubbleState == RUBBLE_BUILDING
+	local isDestroyedMountain = isRubble and rubbleState == RUBBLE_MOUNTAIN
+	
+	if isBuilding or isRuins then
+		self:SetBuilding(loc, hp, hp_max)
+		
+	elseif isMountain or isDestroyedMountain then
+		self:SetMountain(loc, hp, hp_max)
+		
+	elseif iTerrain == TERRAIN_ICE then
+		self:SetIce(loc, hp, hp_max)
+	end
+end
+
+BoardClass.SetMaxHealth = function(self, loc, hp_max)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetMaxHealth",
+		params = { self, loc, hp_max },
+		{ "userdata|GameBoard&", "userdata|Point", "number|int" }
+	}
+	
+	local rubbleState = CUtils.GetTileRubbleState(self, loc)
+	local iTerrain = self:GetTerrain(loc)
+	
+	local isBuilding = iTerrain == TERRAIN_BUILDING
+	local isRuins = iTerrain == TERRAIN_RUBBLE and rubbleState == RUBBLE_BUILDING
+	
+	if isBuilding or isRuins then
+		
+		if self:IsUniqueBuilding(loc) then
+			hp_max = 1
+		else
+			hp_max = math.max(1, math.min(4, hp_max))
+		end
+		
+		local hp = CUtils.GetTileHealth(self, loc.x, loc.y)
+		
+		self:SetBuilding(loc, hp, hp_max)
+	end
+end
```

Comment:
❗ Calling this causes an error on line `local hp = CUtils.GetTileHealth(self, loc.x, loc.y)`, the function expects a `Point`

Reply:
Fixed in `d3d8d13`